src/main/webapp/mapScript.js

```diff
@@ -134,3 +106,34 @@ async function deletePlace(contentId) {
   let URL = "/delete_marker?contentId=" + contentId;
   await fetch(URL, options).then();
 }
+
+async function createRoute() {
+  var routeName = document.getElementById("route-name").value;
+  if (routeName == "") {
+    alert("Please add a name to your new route!");
+  } else {
+    var markersIds = [];
+    for (var i = 0; i < markersArray.length; i++) {
+      markersIds.push({ id: markersArray[i].id });
+    }
+    let options = {
+      method: "POST",
+      body: JSON.stringify(markersIds),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    };
+
+    let URL = "/storeRoute?routeName=" + routeName;
+    await fetch(URL, options).then();
+
+    let tabel = document.getElementById("places-table");
+    tabel.innerHTML = "";
+    for (var i = 0; i < markersArray.length; i++) {
+      markersArray[i].marker.setMap(null);
```

Comment:
It may be better to clear markersArray entirely, so that previous markers won't appear in the new route.

Reply:
That as well causes problem the markers are saved in the datastore several times.

Reply:
Done